ocrd/ocrd/processor/helpers.py

```diff
@@ -95,6 +95,9 @@ def run_processor(
         instance_caching=instance_caching
     )
 
+    old_cwd = getcwd()
```

Comment:
Right. I believe this should be fixed and merged separately to core (#987) since this is a modification to `ocrd` package that could potentially break things in core and not `ocrd-network`. The reason these changes are also reflected in this PR is to be able to run processors. Anyway, I will adopt the change in both PRs.

`TODO don't set this yet!` shall be removed right before merging to core.

Reply:
Yes, in #987 it's the same. Both PRs just **introduced** `old_cwd`, so you are free to move them, nothing can break.

Reply:
In fact, simply moving `old_cwd = getcwd()` before `get_processor()` broke tests. I need to investigate more about why.

Reply:
So you [tried to move getcwd up](https://github.com/OCR-D/core/pull/974/commits/a2a7fe99f2463000f28f5e55e479a456369e83ad), but subsequently [reverted](https://github.com/OCR-D/core/pull/974/commits/d1f287df4300d401473700dff5f2fcbc908e8f2b). Why?

(Unresolving this now.)

Reply:
Because it failed the tests for the dummy processor execution, check [here](https://github.com/OCR-D/core/runs/11714994779). I still did not stack track (as you suggested) to see why it's happening.

Reply:
Ok, I had a look at the 2 CI failures. It looks like actually, the tests are badly constructed!

@kba should weigh in, but my analysis is the following:
1. `TestDummyProcessor.test_copies_ok` create a temporary WS, calls `run_processor` on it, then checks its outputs. One of these tests uses `page_from_file` on the OcrdFile outputs. But naturally, `page_from_file` assumes that we are still in the WS directory – which formally, we are not. But until now, `run_processor` (via `Processor.__init__`) **did chdir to the WS and stayed there**. But if we move the getcwd to before `get_processor`, then the new `finally` clause will undo that. So really, we should rewrite the test to chdir to `wsdir` from the beginning
2. `TestDummyProcessor.test_copy_file_false` ditto.

Reply:
Should be resolved now - both in #987 and here.